enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+enables some simplification of the SNC image build process by
+utilizing the profile to control which components are included in the
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
+node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer=true
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* most operators running in the openshift-monitoring namespace
+* most operators running in the openshift-insights namespace
+* machine-api-operator
+* Machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
+
+Additional components that are not currently removed by the SNC
+tooling might be also be determined to be unnecessary, and thus not
+require the annotation. Possibilities include:
+
+* openshift-cluster-node-tuning-operator
+* openshift-multus
+
+Once PRs adding this annotation have been merged to all the necessary
+component repositories, the snc.sh script will be updated to tell the
+OpenShift installer to use the single-node-developer cluster profile,
+and to remove the code that deletes the components that are not
+annotated. At the conclusion of this initial phase, the somewhat
+simplified CRC build tooling should produce a cluster equivalent to
+that currently produced, and all existing CRC tests should pass. Phase
+one is expected to be completed during development of the 4.7
+OpenShift release.
+
+From this point forward, any new components being added to OpenShift
+will not be included in CRC unless a manifest with the
+single-node-developer annotation is included in the new component. The
+CRC team will need to engage with other OpenShift teams to help decide
+whether or not new components should be included in the profile, and
+if so, how they should be configured.
+
+### Phase Two
+
+In the second phase, various OpenShift components that are included in
+single node developer clusters will be updated to use specialized
+manifests in the single-node-developer profile rather than the same
+manifest used for default clusters. These specialized manifests will
+configure the components for use in single node developer clusters,
+replacing the patching currently done post-deployment by the CRC build
+process, and potentially enabling additional reductions in resource
+requirements.
+
+Some of the manifest specializations expected to be included in this
+phase are shown in the following table:
```

Comment:
who will maintain these manifests?

There are no generic mechanisms in place in many operator to support different operand manifests. This might need a pretty big fan out of work.

Reply:
The expectation is that these manifests would be maintained in each components. The team who introduced the profile could help with this when needed.

Reply:
Once a specialized manifest is committed to a component's repo, that component team will be responsible for maintenance of that manifest. During the 4.7 cycle, I expect the devtools team to propose the patches adding the annotation to existing manifests (phase 1) and adding specialized manifests (phase 2). Going forward, component teams may decide they want to specialize manifests for the single-node-developer profile themselves.